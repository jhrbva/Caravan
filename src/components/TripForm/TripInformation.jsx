import React from 'react';
import { Field } from 'formik';
import '../../utilities/fonts.scss';
import Input from '../Input/Input';
import { required } from '../../utilities/formValidation';

import TitleIcon from '@material-ui/icons/Title';
import DescriptionIcon from '@material-ui/icons/Description';
import TripOriginIcon from '@material-ui/icons/TripOrigin';
import LocationCityIcon from '@material-ui/icons/LocationCity';
import EventIcon from '@material-ui/icons/Event';
import ScheduleIcon from '@material-ui/icons/Schedule';

const TripInformation = () => (
	<>
		<Field
			icon={<TitleIcon />}
			type='text'
			name='trip_title'
			placeholder='Trip Title'
			validate={required}
			autoFocus={true}
			component={Input}
		/>
		<Field
			icon={<DescriptionIcon />}
			type='text'
			name='trip_description'
			placeholder='Description'
			validate={required}
			autoFocus={true}
			component={Input}
		/>
		<Field
			icon={<TripOriginIcon />}
			type='text'
			name='start_location'
			placeholder='Start Location'
			validate={required}
			autoFocus={true}
			component={Input}
		/>
		<Field
			icon={<LocationCityIcon />}
			type='text'
			name='destination'
			placeholder='Destination'
			validate={required}
			autoFocus={false}
			component={Input}
		/>
		<Field
			icon={<EventIcon />}
			type='text'
			name='start_date'
			placeholder='Date'
			validate={required}
			autoFocus={false}
			component={Input}
		/>
		<Field
			icon={<ScheduleIcon />}
			type='text'
			name='start_time'
			placeholder='Time'
			validate={required}
			autoFocus={false}
			component={Input}
		/>
	</>
);

export default TripInformation;
