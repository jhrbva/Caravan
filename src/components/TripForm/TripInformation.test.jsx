import React from 'react';
import { shallow } from 'enzyme';
import TripInformation from './TripInformation';

describe('TripInformation', () => {
	it('should match the snapshot', () => {
		const wrapper = shallow(<TripInformation />);
		expect(wrapper).toMatchSnapshot();
	});
	it('should have four fields', () => {
		const wrapper = shallow(<TripInformation />);
		const fieldWrapper = wrapper.find('Field');
		expect(fieldWrapper.length).toBe(6);
	});
	it('should have an start_location field', () => {
		const wrapper = shallow(<TripInformation />);
		const fieldWrapper = wrapper.find('Field');
		const usernameField = fieldWrapper.find({ name: 'start_location' });
		expect(usernameField.length).toBe(1);
	});
	it('should have a destination field', () => {
		const wrapper = shallow(<TripInformation />);
		const fieldWrapper = wrapper.find('Field');
		const usernameField = fieldWrapper.find({ name: 'destination' });
		expect(usernameField.length).toBe(1);
	});
	it('should have an start_date field', () => {
		const wrapper = shallow(<TripInformation />);
		const fieldWrapper = wrapper.find('Field');
		const usernameField = fieldWrapper.find({ name: 'start_date' });
		expect(usernameField.length).toBe(1);
	});
	it('should have a start_time field', () => {
		const wrapper = shallow(<TripInformation />);
		const fieldWrapper = wrapper.find('Field');
		const usernameField = fieldWrapper.find({ name: 'start_time' });
		expect(usernameField.length).toBe(1);
	});
});
