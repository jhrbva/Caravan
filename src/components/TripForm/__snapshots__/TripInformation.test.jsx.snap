// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TripInformation should match the snapshot 1`] = `
<Form>
  <Field
    autoFocus={true}
    component={[Function]}
    icon="trip_origin"
    name="start_location"
    placeholder="Start Location"
    type="text"
    validate={[Function]}
  />
  <Field
    autoFocus={false}
    component={[Function]}
    icon="location_city"
    name="destination"
    placeholder="Destination"
    type="text"
    validate={[Function]}
  />
  <Field
    autoFocus={false}
    component={[Function]}
    icon="event"
    name="start_date"
    placeholder="Date"
    type="text"
    validate={[Function]}
  />
  <Field
    autoFocus={false}
    component={[Function]}
    icon="schedule"
    name="start_time"
    placeholder="Time"
    type="text"
    validate={[Function]}
  />
  <BigButton
    value="Next"
  />
</Form>
`;
